indico/modules/events/editing/client/js/components/EditableTypeList.jsx

```diff
@@ -0,0 +1,126 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import editableTypesURL from 'indico-url:event_editing.api_editable_types';
+
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {Button, Checkbox, Form, Loader} from 'semantic-ui-react';
+import {Form as FinalForm} from 'react-final-form';
+import {Param, Translate} from 'indico/react/i18n';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {FinalCheckbox, FinalSubmitButton, handleSubmitError} from 'indico/react/forms';
+import {indicoAxios} from 'indico/utils/axios';
+
+import './EditableTypeList.module.scss';
+
+export default function EditableTypeList({eventId}) {
+  const [editMode, setEditMode] = useState(false);
+  const {data: editableTypes, loading: isLoadingEditableTypes, reFetch} = useIndicoAxios({
+    url: editableTypesURL({confId: eventId}),
+    camelize: true,
+    trigger: eventId,
+  });
+
+  if (isLoadingEditableTypes) {
+    return <Loader inline="centered" active />;
+  } else if (!editableTypes) {
+    return null;
+  }
+
+  const handleSubmit = async formData => {
+    const url = editableTypesURL({confId: eventId});
+    const enabledTypes = Object.keys(formData).filter(name => formData[name]);
+    try {
+      await indicoAxios.post(url, {editable_types: enabledTypes});
+      reFetch();
+    } catch (e) {
+      return handleSubmitError(e);
+    }
+    setEditMode(false);
+  };
+
+  return (
+    <div styleName="editable-list">
+      <h3>List of editable types</h3>
+      <FinalForm
+        onSubmit={handleSubmit}
+        subscription={{submitting: true}}
+        initialValues={Object.assign(...editableTypes.map(([id, {enabled}]) => ({[id]: enabled})))}
+      >
+        {fprops => (
+          <>
+            <Form id="editable-type-form" onSubmit={fprops.handleSubmit}>
+              <div className="action-box">
+                {!editMode &&
```

Comment:
I'd keep the two modes more separate, i.e. render the FinalForm and the form content itself only while in edit mode, and the normal display version only while outside edit mode. The duplication would be minimal (some contaienr elements), but I think it would greatly improve readability.